packages/react-components/src/components/resource-explorers/types/request-fn.ts

```diff
@@ -0,0 +1,144 @@
+import type {
+  BatchGetAssetPropertyValueRequest,
+  BatchGetAssetPropertyValueResponse,
+  ExecuteQueryRequest,
+  ExecuteQueryResponse,
+  ListAssetModelsRequest,
+  ListAssetModelsResponse,
+  ListAssetModelPropertiesRequest,
+  ListAssetModelPropertiesResponse,
+  ListAssetsRequest,
+  ListAssetsResponse,
+  ListAssociatedAssetsRequest,
+  ListAssociatedAssetsResponse,
+  ListAssetPropertiesRequest,
+  ListAssetPropertiesResponse,
+  ListTimeSeriesRequest,
+  ListTimeSeriesResponse,
+} from '@aws-sdk/client-iotsitewise';
+import type { RequestTimeout } from './common';
+
+/** First-class function used to send requests to AWS. */
+export type RequestFunction<Request, Response> = (
+  request: Request,
+  options?: {
+    abortSignal?: AbortSignal;
+    requestTimeout?: RequestTimeout;
+  }
+) => PromiseLike<Response>;
+
+/** Utility type to ensure the correct structure of `requestFns`. */
+export type RequestFunctions<T> = T extends Partial<
```

Comment:
Not saying we should move these as part of this PR, but these might be useful types to have for other systems that need to make requests. Maybe it should be in a common types in react-components?

Reply:
Completely agree. We should have all the components use these types (where it makes sense, of course).